nixos/modules/misc/mindustry.nix

```
@@ -0,0 +1,41 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+with lib; let
+  cfg = config.services.mindustry-server;
+in {
+  options = {
+    services.mindustry-server = {
+      enable = mkEnableOption "Mindustry server";
+
+      package = mkPackageOption pkgs "mindustry-server" {};
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Open ports in the firewall for the mindustry server";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.mindustry = {
+      enable = true;
+      description = "Mindustry server";
+      serviceConfig = {
+        Type = "simple";
+        ExecStart = "${pkgs.mindustry-server}/bin/mindustry-server host";
+        ExecStop = "pkill server-release.jar";
+      };
```

This would make the service run as root, I'd suggest at least enabling `DynamicUser`:

```suggestion
      serviceConfig = {
        Type = "simple";
        ExecStart = "${pkgs.mindustry-server}/bin/mindustry-server host";
        ExecStop = "pkill server-release.jar";
        DynamicUser = true;
      };
```